### PR TITLE
Fix Pipeline stuck for INSERT SELECT FINAL

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix `Pipeline stuck` exception for `INSERT SELECT FINAL` where `SELECT` (`max_threads`>1) has multiple streams but `INSERT` has only one (`max_insert_threads`==0)

Cc: @KochetovNicolai (please take a look)

### CODE REVIEW

File: src/Processors/ConcatProcessor.cpp [hunk 1]
```diff
@@ -4,6 +4,17 @@
 namespace DB
 {
 
+ConcatProcessor::ConcatProcessor(const Block & header, size_t num_inputs)
+    : IProcessor(InputPorts(num_inputs, header), OutputPorts{header}), current_input(inputs.begin())
+{
+}
+
+void ConcatProcessor::prepareInitializeInputs()
```
Comment (KochetovNicolai) [inline src/Processors/ConcatProcessor.cpp:12]: I'm afraid this brakes pretty important invariant of `ConcatProcessor`: we don't activate any input until it's turn to read from. I will improve comment.

**Review by KochetovNicolai — CHANGES_REQUESTED** [review]

File: src/Processors/ConcatProcessor.cpp (hunk 1 above)
Comment (KochetovNicolai) [reply src/Processors/ConcatProcessor.cpp:12]: But, generally, you right. Let's just replace `ConcatProcessor` to `Resize`.

File: src/Interpreters/InterpreterInsertQuery.cpp [hunk 2]
```diff
@@ -209,10 +209,7 @@ BlockIO InterpreterInsertQuery::execute()
             if (table->supportsParallelInsert() && settings.max_insert_threads > 1)
                 out_streams_size = std::min(size_t(settings.max_insert_threads), res.pipeline.getNumStreams());
 
-            if (out_streams_size == 1)
-                res.pipeline.addPipe({std::make_shared<ConcatProcessor>(res.pipeline.getHeader(), res.pipeline.getNumStreams())});
-            else
-                res.pipeline.resize(out_streams_size);
+            res.pipeline.resize(out_streams_size);
```
Comment (azat) [inline src/Interpreters/InterpreterInsertQuery.cpp:212]: Nice idea, didn't though about this

**Review by KochetovNicolai — APPROVED** [review]

**Comment by KochetovNicolai** [comment]
No backport because there are no affected releases.

**Pull request merged**